stdlib/public/core/IntegerParsing.swift

```diff
@@ -197,4 +197,31 @@ extension FixedWidthInteger {
   public init?(_ description: String) {
     self.init(description, radix: 10)
   }
+  
+  /// Creates a new integer value from the given Substring.
+  ///
+  /// Substring can be parsed at a high speed just by substituting strings.
+  ///
+  /// The Substring passed as `description` may begin with a plus or minus sign
+  /// character (`+` or `-`), followed by one or more numeric digits (`0-9`).
+  ///
+  ///     let x = Int("123")
+  ///     // x == 123
+  ///
+  /// If `description` is in an invalid format, or if the value it denotes in
+  /// base 10 is not representable, the result is `nil`. For example, the
+  /// following conversions result in `nil`:
+  ///
+  ///     Int(" 100")                       // Includes whitespace
+  ///     Int("21-50")                      // Invalid format
+  ///     Int("ff6600")                     // Characters out of bounds
+  ///     Int("10000000000000000000000000") // Out of range
+  ///
+  /// - Parameter description: The ASCII representation of a number.
+  @inlinable
```

Comment:
This needs availability annotations for the standard library, and probably `@_alwaysEmitIntoClient` for ABI reasons. I forget exactly, but I think that implies `@inlinable`.